server/tserver/src/main/java/org/apache/accumulo/tserver/log/RecoveryLogsIterator.java

```diff
@@ -79,7 +86,8 @@ public RecoveryLogsIterator(ServerContext context, List<Path> recoveryLogDirs, L
 
       for (Path log : logFiles) {
         var scanner = RFile.newScanner().from(log.toString()).withFileSystem(fs)
-            .withTableProperties(context.getConfiguration()).build();
+            .withTableProperties(context.getConfiguration()).withDataCache(dataCacheSize)
```

Comment:
This cache is per recovery log iterator.  This does not help with the case where tablet server has 100 tablets that need to recover data from the same walogs.  To help with this case would need to create dedicated block caches for recovery at the tsever level and pass them into this iterator.  Probably would want their own config so they can be size differently than the block caches for scans.  Also probably would not want to use the scan block cache for this as it will always be disjoint files and we probably do not want recovery to impact the caches used for scans by causing evictions.

Reply:
OK that is what I thought. I thought this was at least better than what we have now.

> would not want to use the scan block cache for this

What do you mean?

Reply:
So do you think we would need a separate Data cache and index cache dedicated to Recovery? Then have the TabletServerResourceManager return a dedicated ScanCacheProvider for recovery?

Reply:
So I created a separate BlockCache for Recovery in the TabletServer but once I pass it down to the Recovery code, I am having trouble creating an Iterator from the `FileSKVIterator` type. The `FileSKVIterator` type allows passing in a `CacheProvider` but then why I try to convert the iterator to a generic java iterator I get stuck. For the new Recovery code we are using Scanners to iterate over the R-files and keep track of this by creating a `Iterator<Entry<Key,Value>> iter` from calling the `iterator()` method.
https://github.com/apache/accumulo/blob/1c0aa8f54642d42c5c4fbb6970141b8acd7a63d9/core/src/main/java/org/apache/accumulo/core/client/ScannerBase.java#L198

That method does the work of creating caches but I don't have access to them in `RFileScanner`.
https://github.com/apache/accumulo/blob/7d176b60fc6a2ce28fcc3cfdc9a3b68684c76f87/core/src/main/java/org/apache/accumulo/core/client/rfile/RFileScanner.java#L340

Any ideas?